Unprocessable Entity when trying to update check

GREAT action!  So I'm trying to use it and the Rubocop command is running and reporting violations but it looks like it's failing when trying to update the check:

```
/action/lib/index.rb:63:in `update_check': Unprocessable Entity (RuntimeError)
	from /action/lib/index.rb:125:in `run'
	from /action/lib/index.rb:132:in `<main>'
```

I'm wondering if it's because I'm trying to use this on a private repo?  The github token is mapped in the env though so it SHOULD have permissions...

Hi.  Here's the full output:'

```
Run andrewmcodes/rubocop-linter-action@v0.1.2
/usr/bin/docker run --name fbc64ffec87db3c49b4c40817119b46fbc6a8e_26de33 --label fbc64f --workdir /github/workspace --rm -e GITHUB_TOKEN -e HOME -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e GITHUB_ACTIONS=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/actions-test/actions-test":"/github/workspace" fbc64f:fec87db3c49b4c40817119b46fbc6a8e
Building native extensions. This could take a while...
Successfully installed jaro_winkler-1.5.3
Successfully installed parallel-1.18.0
Successfully installed ast-2.4.0
Successfully installed parser-2.6.5.0
Successfully installed rainbow-3.0.0
Successfully installed ruby-progressbar-1.10.1
Successfully installed unicode-display_width-1.6.0
Successfully installed rubocop-0.75.1
Successfully installed rubocop-performance-1.5.0
Successfully installed rack-2.0.7
Successfully installed rubocop-rails-2.3.2
11 gems installed
/github/workspace/app/models/shopify/account.rb: Warning: no department given for HandleExceptions. Run `rubocop -a --only Migration/DepartmentName` to fix.
/github/workspace/app/models/shopify/account.rb: Warning: no department given for HandleExceptions. Run `rubocop -a --only Migration/DepartmentName` to fix.
/github/workspace/app/models/subscription/stripe.rb: Warning: no department given for DuplicateMethods. Run `rubocop -a --only Migration/DepartmentName` to fix.
/github/workspace/app/models/subscription/stripe.rb: Warning: no department given for DuplicateMethods. Run `rubocop -a --only Migration/DepartmentName` to fix.
/action/lib/index.rb:63:in `update_check': Unprocessable Entity (RuntimeError)
	from /action/lib/index.rb:125:in `run'
	from /action/lib/index.rb:132:in `<main>'
##[error]Docker run failed with exit code 1
```

and it's using the `setup-ruby` action which I'm using 2.6.3

```
 Set up Ruby 2.63s
##[add-path]/opt/hostedtoolcache/Ruby/2.6.3/x64/bin
Run actions/setup-ruby@v1
sudo ln -sf /opt/hostedtoolcache/Ruby/2.6.3/x64/bin/ruby /usr/bin/ruby
##[add-path]/opt/hostedtoolcache/Ruby/2.6.3/x64/bin
```

Alas, the repo is a private repo..and I'm wondering if that is something to do with it?

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,6 @@
-FROM ruby:2.6
+FROM ruby:2.6.5-alpine
+
+RUN apk add --update build-base git
 
 LABEL com.github.actions.name="Rubocop Linter"
 LABEL com.github.actions.description="Lint your Ruby code"
@@ -8,4 +10,6 @@ LABEL com.github.actions.color="red"
 LABEL maintainer="Andrew Mason <andrewmcodes@protonmail.com>"
 
 COPY lib /action/lib
+RUN gem install bundler
+
 ENTRYPOINT ["/action/lib/entrypoint.sh"]

--- a/README.md
+++ b/README.md
@@ -2,9 +2,11 @@
 
 # Rubocop Linter Action
 
-![version number](https://img.shields.io/static/v1?label=Version&message=v0.1.2&color=blue)
+![version number](https://img.shields.io/static/v1?label=Version&message=v0.2.0&color=blue)
 
-GitHub Action to run Rubocop against your code.
+GitHub Action to run Rubocop against your code and create annotations in the UI.
+
+**NOTE: due to the GitHub Check Runs API, we can only return 50 annotations per run. See [here](https://developer.github.com/v3/checks/runs/#output-object) for more info.**
 
 ## Usage
 
@@ -12,7 +14,7 @@ Add the following to your GitHub action workflow:
 
 ```yaml
 - name: Rubocop Linter
-  uses: andrewmcodes/rubocop-linter-action@v0.1.2
+  uses: andrewmcodes/rubocop-linter-action@v0.2.0
   env:
     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 ```
@@ -20,27 +22,23 @@ Add the following to your GitHub action workflow:
 ## Example Workflow
 
 ```yaml
-name: Ruby
+name: Rubocop
 
-on: [push]
+on:
+  pull_request:
+    branches:
+      - '*'
+  push:
+    branches:
+      - master
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
-
     steps:
     - uses: actions/checkout@v1
-    - name: Set up Ruby 2.6
-      uses: actions/setup-ruby@v1
-      with:
-        ruby-version: 2.6.x
-    - name: add PostgreSQL dependencies
-      run: |
-        sudo apt-get update
-        sudo apt-get install -y postgresql-client libpq-dev
     - name: Rubocop Linter
-      uses: andrewmcodes/rubocop-linter-action@v0.1.2
+      uses: andrewmcodes/rubocop-linter-action@v0.2.0
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 ```

--- a/lib/entrypoint.sh
+++ b/lib/entrypoint.sh
@@ -2,6 +2,6 @@
 
 set -e
 
-gem install rubocop rubocop-performance rubocop-rails
+gem install rubocop rubocop-performance rubocop-rails rubocop-minitest rubocop-rspec
 
 ruby /action/lib/index.rb

--- a/lib/index.rb
+++ b/lib/index.rb
@@ -11,9 +11,8 @@ require 'time'
 @repository = @event["repository"]
 @owner = @repository["owner"]["login"]
 @repo = @repository["name"]
-
+@conclusion = "success"
 @check_name = "Rubocop"
-
 @headers = {
   "Content-Type": 'application/json',
   "Accept": 'application/vnd.github.antiope-preview+json',
@@ -46,8 +45,6 @@ end
 def update_check(id, conclusion, output)
   body = {
     "name" => @check_name,
-    "head_sha" => @GITHUB_SHA,
-    "status" => 'completed',
     "completed_at" => Time.now.iso8601,
     "conclusion" => conclusion,
     "output" => output
@@ -56,7 +53,6 @@ def update_check(id, conclusion, output)
   http = Net::HTTP.new('api.github.com', 443)
   http.use_ssl = true
   path = "/repos/#{@owner}/#{@repo}/check-runs/#{id}"
-
   resp = http.patch(path, body.to_json, @headers)
 
   if resp.code.to_i >= 300
@@ -73,13 +69,25 @@ end
 }
 
 def run_rubocop
-  annotations = []
+  annotations = annotation_messages
+
+  output = {
+    "title": @check_name,
+    "summary": "#{annotations.size} offense(s) found",
+    "annotations" => annotations
+  }
+
+  return { "output" => output, "conclusion" => @conclusion }
+end
+
+def annotation_messages
   errors = nil
+  count = 0
+  annotation_list = []
+
   Dir.chdir(@GITHUB_WORKSPACE) {
     errors = JSON.parse(`rubocop --format json`)
   }
-  conclusion = "success"
-  count = 0
 
   errors["files"].each do |file|
     path = file["path"]
@@ -92,27 +100,22 @@ def run_rubocop
       annotation_level = @annotation_levels[severity]
       count = count + 1
 
+      return annotation_list if count == 48
+
       if annotation_level == "failure"
-        conclusion = "failure"
+        @conclusion = "failure"
       end
 
-      annotations.push({
-                         "path" => path,
-                         "start_line" => location["start_line"],
-                         "end_line" => location["start_line"],
-                         "annotation_level": annotation_level,
-                         "message" => message
-                       })
+      annotation_list.push({
+                        "path" => path,
+                        "start_line" => location["start_line"],
+                        "end_line" => location["start_line"],
+                        "annotation_level": annotation_level,
+                        "message" => message
+                      })
     end
   end
-
-  output = {
-    "title": @check_name,
-    "summary": "#{count} offense(s) found",
-    "annotations" => annotations
-  }
-
-  return { "output" => output, "conclusion" => conclusion }
+  return annotation_list
 end
 
 def run
@@ -121,7 +124,6 @@ def run
     results = run_rubocop()
     conclusion = results["conclusion"]
     output = results["output"]
-
     update_check(id, conclusion, output)
   rescue
     update_check(id, "failure", nil)

